# NAILDOH: Social Class Part 1

## Resources

In [374]:
# Libraries
library(tidyverse) # for data manipulation

In [375]:
# Functions
factorize <- function(df){ # Create a function
  for(i in which(sapply(df, class) == "character")) # that looks for variables with the character class 
      df[[i]] = as.factor(df[[i]]) # and converts them to factor (i.e., categorical) class
  return(df)
}

unfactorize <- function(df){ # Create a function
  for(i in which(sapply(df, class) == "factor")) # that looks for variables with the character class 
      df[[i]] = as.character(df[[i]]) # and converts them to factor (i.e., categorical) class
  return(df)
}

In [376]:
# Data
letters <- factorize(read.csv("20230218_AM_PhD-NaildohSubset.csv")) # Put csv into a dataframe called docData
colnames(letters) # Get an overview of the dataframe
dim(letters)

[1] "docid"                     "sourcetitle"              
 [3] "docyear"                   "docmonth"                 
 [5] "docday"                    "authorLocation"           
 [7] "docauthorid"               "docauthorname"            
 [9] "authorgender"              "agewriting"               
[11] "birthyear"                 "deathyear"                
[13] "religion"                  "cultural_heritage"        
[15] "north_american_occupation" "native_occupation"        
[17] "marriagestatus"            "maternalstatus"           
[19] "nationalOrigin"            "britishEmpire_EU"         
[21] "translated"                "publicLetter"             
[23] "religionNew"               "relMin"

[1] 617  24

In [377]:
# Compare the two variables values and proportion of missing data
unique(letters$north_american_occupation)
sum(is.na(letters$north_american_occupation))/617

unique(letters$native_occupation)
sum(is.na(letters$native_occupation))/617

[1] Nun; Social worker; Teacher                                       
 [2] Military wife; Writer                                             
 [3] <NA>                                                              
 [4] Clergy                                                            
 [5] Farmer                                                            
 [6] Farmer's wife                                                     
 [7] Miner                                                             
 [8] Homemaker                                                         
 [9] Architect; Artist; Businessman; Clergy; Educator                  
[10] Architect; Artist; Businessman; Clergy                            
[11] Architect; Artist; Businessman; Clergy; Educator; Farmer          
[12] Architect; Artist; Businessman; Educator; Farmer                  
[13] Architect; Artist; Businessman; Clergy; Educator; Farmer; Merchant
[14] Government appointee; Politician; Religious leader                
[15] Farmer; Tradesman                                                 
[16] Miner; Rancher; Laborer                                           
[17] Clergy; Educator; Writer                                          
[18] Government employee                                               
[19] Government appointee                                              
[20] Teacher                                                           
[21] Tradesman                                                         
[22] Surveyor                                                          
[23] Manufacturer                                                      
[24] Clergy; Missionary                                                
[25] Royal governor                                                    
[26] Laborer; Tradesman                                                
[27] Writer                                                            
26 Levels: Architect; Artist; Businessman; Clergy ...

[1] 0.1296596

[1] <NA>                          Clergy                       
 [3] Clergy; Farmer                Businessman; Farmer; Merchant
 [5] Businessman; Educator         Farmer                       
 [7] Farmer; Tradesman             Businessman; Tradesman       
 [9] Teacher                       Tradesman                    
[11] Urban planner                 Manufacturer                 
[13] Military personnel            Writer                       
13 Levels: Businessman; Educator ... Writer

[1] 0.4959481

In [378]:
# There is a higher occurance of NA for native_occupation
# So I will use the north_american_occupation as a starting point.
# When native_occupation is available, what is the north_american_occupation value?
letters  %>% 
filter(!is.na(native_occupation)) %>%
select(docauthorname, native_occupation, north_american_occupation)  %>% 
unique()

docauthorname                                             
1   Chidlaw, Benjamin Williams, 1811-1892                     
2   Roberts, Samuel, fl. 1856-1870                            
4   Harris, Critchlow, 1813-1899                              
5   Harris, Sarah Stretch, 1818-1897                          
44  Harris, Tom, 1847-                                        
54  Harris, Robert, 1849-1919                                 
84  Harris, Margaret Ellin, 1851-1944                         
179 Harris, William, 1854-                                    
231 Carrothers, Nathaniel, ?-1881                             
238 Carrothers, Joseph, 1793(?)-                              
249 Robb, Alexander, 1839-                                    
261 Bishop, Robert Hamilton, 1777-1855                        
262 Bell, William, 1780-1857                                  
288 Knight, James, fl. 1831                                   
289 Menzies, George, fl. 1834                                 
291 Anonymous Male Scottish Farmer Immigrant, fl. 1833        
295 Anonymous Scottish Clergyman in Canada, fl. 1832          
296 Anonymous Scottish Cabinet Maker, fl. 1833                
297 Anonymous Scottish Gardener in Canada, fl. 1834           
298 Anonymous Scottish Farmer from St. Fergus Parish, fl. 1834
299 Anonymous Scottish Immigrant from Turriff, fl. 1834       
300 McKenzie, D., fl. 1834                                    
301 Anonymous Scottish Farmer from Aberdeenshire, fl. 1834    
302 Aylmer, Matthew, Lord, 1775-1850                          
303 Singer, William, fl. 1831                                 
304 Mooney, Thomas, fl. 1850                                  
307 Hutchings, James Mason, 1820-1902                         
311 Downe, John, fl. 1830                                     
    native_occupation            
1   Clergy                       
2   Clergy; Farmer               
4   Businessman; Farmer; Merchant
5   Businessman; Educator        
44  Businessman; Farmer; Merchant
54  Businessman; Farmer; Merchant
84  Businessman; Farmer; Merchant
179 Businessman; Farmer; Merchant
231 Farmer                       
238 Farmer; Tradesman            
249 Farmer                       
261 Clergy                       
262 Clergy                       
288 Businessman; Tradesman       
289 Teacher                      
291 Farmer                       
295 Clergy                       
296 Tradesman                    
297 Urban planner                
298 Farmer                       
299 Manufacturer                 
300 Clergy                       
301 Farmer                       
302 Military personnel           
303 Tradesman                    
304 Writer                       
307 Tradesman                    
311 Tradesman                    
    north_american_occupation                                         
1   Clergy                                                            
2   NA                                                                
4   Architect; Artist; Businessman; Clergy; Educator                  
5   Architect; Artist; Businessman; Clergy                            
44  Architect; Artist; Businessman; Clergy; Educator; Farmer          
54  Architect; Artist; Businessman; Clergy; Educator; Farmer          
84  Architect; Artist; Businessman; Educator; Farmer                  
179 Architect; Artist; Businessman; Clergy; Educator; Farmer; Merchant
231 Farmer; Tradesman                                                 
238 Farmer; Tradesman                                                 
249 Miner; Rancher; Laborer                                           
261 Clergy; Educator; Writer                                          
262 Clergy                                                            
288 NA                                                                
289 Teacher                                                           
291 Farmer                                             

Some of these values match. At this point, I am concerned with the cases that do not match, either because north_american_occupation is missing or different from native_occupation.

In [379]:
# Get the rows where
rows = which(!is.na(letters$native_occupation) & # Native occupation is not NA
             is.na(letters$north_american_occupation) | # the North American occupation is NA or...
             # the native and North American occupations differ
             as.character(letters$native_occupation) != as.character(letters$north_american_occupation))

In [380]:
# Now...
letters %>% #in the dataframe
filter(row_number() %in% rows) %>% # in the rows that meet those conditions
select(docauthorname, native_occupation, north_american_occupation) %>% # view values for these variables
unique() # only once where they occur together

docauthorname                                      
1   Roberts, Samuel, fl. 1856-1870                     
2   Harris, Critchlow, 1813-1899                       
3   Harris, Sarah Stretch, 1818-1897                   
42  Harris, Tom, 1847-                                 
52  Harris, Robert, 1849-1919                          
82  Harris, Margaret Ellin, 1851-1944                  
177 Harris, William, 1854-                             
229 Carrothers, Nathaniel, ?-1881                      
236 Robb, Alexander, 1839-                             
248 Bishop, Robert Hamilton, 1777-1855                 
249 Knight, James, fl. 1831                            
250 Anonymous Scottish Gardener in Canada, fl. 1834    
251 Anonymous Scottish Immigrant from Turriff, fl. 1834
252 McKenzie, D., fl. 1834                             
253 Aylmer, Matthew, Lord, 1775-1850                   
254 Singer, William, fl. 1831                          
255 Hutchings, James Mason, 1820-1902                  
259 Downe, John, fl. 1830                              
    native_occupation            
1   Clergy; Farmer               
2   Businessman; Farmer; Merchant
3   Businessman; Educator        
42  Businessman; Farmer; Merchant
52  Businessman; Farmer; Merchant
82  Businessman; Farmer; Merchant
177 Businessman; Farmer; Merchant
229 Farmer                       
236 Farmer                       
248 Clergy                       
249 Businessman; Tradesman       
250 Urban planner                
251 Manufacturer                 
252 Clergy                       
253 Military personnel           
254 Tradesman                    
255 Tradesman                    
259 Tradesman                    
    north_american_occupation                                         
1   NA                                                                
2   Architect; Artist; Businessman; Clergy; Educator                  
3   Architect; Artist; Businessman; Clergy                            
42  Architect; Artist; Businessman; Clergy; Educator; Farmer          
52  Architect; Artist; Businessman; Clergy; Educator; Farmer          
82  Architect; Artist; Businessman; Educator; Farmer                  
177 Architect; Artist; Businessman; Clergy; Educator; Farmer; Merchant
229 Farmer; Tradesman                                                 
236 Miner; Rancher; Laborer                                           
248 Clergy; Educator; Writer                                          
249 NA                                                                
250 NA                                                                
251 NA                                                                
252 Clergy; Missionary                                                
253 Royal governor                                                    
254 Laborer; Tradesman                                                
255 Miner                                                             
259 Manufacturer

In [381]:
# Create a new variable called occupation
# Change to character class for transformation
letters$occupation  <- as.character(letters$north_american_occupation)

In [382]:
# Get the rows where
rows = which(!is.na(letters$native_occupation) & # Native occupation is not NA
             is.na(letters$north_american_occupation)) #North American occupation is NA

letters$occupation[rows]

[1] NA NA NA NA

In [383]:
# Recode occupation for those rows to match native occupation.
letters$occupation[rows] <- as.character(letters$native_occupation[rows]) # Recode data
letters$occupation[rows]

[1] "Clergy; Farmer"         "Businessman; Tradesman" "Urban planner"         
[4] "Manufacturer"

Now I have to decide whether to combine jobs from native and North American occupations into one variable. Because there are only a few cases where this is a problem, I can make decisions on a case-by-case basis.

In [384]:
# Get the rows where
rows = which(!is.na(letters$native_occupation) & # Native occupation is not NA
             as.character(letters$native_occupation) != as.character(letters$north_american_occupation))

# Now...
letters %>% #in the dataframe
filter(row_number() %in% rows) %>% # in the rows that meet those conditions
select(docauthorname, native_occupation, north_american_occupation) %>% # view values for these variables
unique() # only once where they occur together

docauthorname                      native_occupation            
1   Harris, Critchlow, 1813-1899       Businessman; Farmer; Merchant
2   Harris, Sarah Stretch, 1818-1897   Businessman; Educator        
41  Harris, Tom, 1847-                 Businessman; Farmer; Merchant
51  Harris, Robert, 1849-1919          Businessman; Farmer; Merchant
81  Harris, Margaret Ellin, 1851-1944  Businessman; Farmer; Merchant
176 Harris, William, 1854-             Businessman; Farmer; Merchant
228 Carrothers, Nathaniel, ?-1881      Farmer                       
235 Robb, Alexander, 1839-             Farmer                       
247 Bishop, Robert Hamilton, 1777-1855 Clergy                       
248 McKenzie, D., fl. 1834             Clergy                       
249 Aylmer, Matthew, Lord, 1775-1850   Military personnel           
250 Singer, William, fl. 1831          Tradesman                    
251 Hutchings, James Mason, 1820-1902  Tradesman                    
255 Downe, John, fl. 1830              Tradesman                    
    north_american_occupation                                         
1   Architect; Artist; Businessman; Clergy; Educator                  
2   Architect; Artist; Businessman; Clergy                            
41  Architect; Artist; Businessman; Clergy; Educator; Farmer          
51  Architect; Artist; Businessman; Clergy; Educator; Farmer          
81  Architect; Artist; Businessman; Educator; Farmer                  
176 Architect; Artist; Businessman; Clergy; Educator; Farmer; Merchant
228 Farmer; Tradesman                                                 
235 Miner; Rancher; Laborer                                           
247 Clergy; Educator; Writer                                          
248 Clergy; Missionary                                                
249 Royal governor                                                    
250 Laborer; Tradesman                                                
251 Miner                                                             
255 Manufacturer

No action is needed for lines 228, 247, 248 and 250 because the value in native_occupation is included in north_american_occupation.

In [385]:
# Biographical research on the Harris (lines 1, 2, 41, 51, 81 and 176) lead to the following recoding.

# Recode according to http://sarah.emilieroberts.ca/index.php?page=The_Family
letters$occupation[letters$docauthorname=="Harris, Critchlow, 1813-1899"]  <- "Farmer; Merchant; Laborer"
letters$occupation[letters$docauthorname=="Harris, Sarah Stretch, 1818-1897"]  <- "Farmer; Merchant; Laborer"
letters$occupation[letters$docauthorname=="Harris, Tom, 1847-"]  <- "Businessman"

#http://www.biographi.ca/en/bio/harris_robert_14E.html
letters$occupation[letters$docauthorname=="Harris, Robert, 1849-1919"]  <- "Artist; Educator"

#http://www.biographi.ca/en/bio/harris_william_critchlow_14E.html
letters$occupation[letters$docauthorname=="Harris, William, 1854-"]  <- "Architect"


#http://www.biographi.ca/en/bio/cotton_william_lawson_15E.html
letters$occupation[letters$docauthorname=="Harris, Margaret Ellin, 1851-1944"]  <- "Writer; Printer"

In [386]:
# Line 235
letters  %>% 
filter(docauthorname=="Robb, Alexander, 1839-") %>%
select(nationalOrigin, religion, religionNew, relMin)  %>% 
unique()

nationalOrigin religion religionNew relMin
1 Irish          NA       NA          NA

In [387]:
#Biographical reseasrch yielded religion and confirmed that at the time of writing, occupation was as shown in 
#north_american_occupation is correct but missing his role as a public servant (supervisor)
#https://www.ancestryireland.com/alexander-robb-canada/
#Also found that the writer was Ulster, which would make him a Protestant, but the denomination is unknown. 
#https://en.wikipedia.org/wiki/Ulster_Scots_people
#"...Robb, scion of a middle-class Protestant Irish family, had finally found a full-time job as a pioneer rancher..." (S9635.txt)
letters$occupation[letters$docauthorname=="Robb, Alexander, 1839-"]  <- "Miner; Laborer; Rancher; Government employee"

#Check
letters  %>% 
filter(docauthorname=="Robb, Alexander, 1839-") %>%
select(nationalOrigin, relMin, occupation)  %>% 
unique()

nationalOrigin relMin occupation                                  
1 Irish          NA     Miner; Laborer; Rancher; Government employee

In [388]:
# Line 249
letters  %>% 
filter(docauthorname=="Aylmer, Matthew, Lord, 1775-1850") %>%
select(docid, nationalOrigin, religion, religionNew, relMin)  %>% 
unique()

docid      nationalOrigin religion religionNew relMin
1 S9865-D072 English        NA       NA          NA

On closer inspection, this is an excerpt of a letter written in an official capacity and presented in a public forum (House of Commons). 

In [389]:
# Note this in case it makes sense to omit public letters at a later stage.
letters$publicLetter[letters$docauthorname=="Aylmer, Matthew, Lord, 1775-1850"] <- TRUE

In [390]:
# Matthew Aylmer: "army officer and colonial administrator"
# "a devout member of the Church of England, Aylmer was unusually tolerant in his religious views"
# http://www.biographi.ca/en/bio/whitworth_aylmer_matthew_7E.html
letters$occupation[letters$docauthorname == "Aylmer, Matthew, Lord, 1775-1850"]  <- 
"Military personnel; Royal governor"
letters$religionNew[letters$docauthorname=="Aylmer, Matthew, Lord, 1775-1850"] <- "Anglican"
letters$relMin[letters$docauthorname=="Aylmer, Matthew, Lord, 1775-1850"] <- FALSE

#Check
letters  %>% 
filter(docauthorname=="Aylmer, Matthew, Lord, 1775-1850") %>%
select(occupation, religionNew, relMin, publicLetter)  %>% 
unique()

occupation                         religionNew relMin publicLetter
1 Military personnel; Royal governor Anglican    FALSE  TRUE

In [391]:
# Line 251
letters  %>% 
filter(docauthorname=="Hutchings, James Mason, 1820-1902") %>%
select(docid, nationalOrigin, religionNew, relMin, authorLocation, native_occupation, north_american_occupation, publicLetter)  %>% 
unique()

docid      nationalOrigin religionNew  relMin authorLocation
1 S9957-D012 English        Episcopalian FALSE  USA           
2 S9957-D013 English        Episcopalian FALSE  USA           
3 S9957-D014 English        Episcopalian FALSE  USA           
4 S9957-D015 English        Episcopalian FALSE  USA           
  native_occupation north_american_occupation publicLetter
1 Tradesman         Miner                     NA          
2 Tradesman         Miner                     NA          
3 Tradesman         Miner                     NA          
4 Tradesman         Miner                     NA

In [392]:
# From S9957
# "journalist, publisher, Yosemite pioneer and inn-keeper"
# "to learn and practice the family trade of cabinet making and general carpentry"
# "it was his carpentry skill which altered his life on the trail from that of a plodding emigrant to a well-paid, high-riding employee of an Army "train.""
# https://en.wikipedia.org/wiki/James_Mason_Hutchings

letters$occupation[letters$docauthorname == "Hutchings, James Mason, 1820-1902"]  <- 
"Tradesman; Miner; Writer; Businessman"

# "letter printed in a New Orleans paper" (S9957)
letters$publicLetter[letters$docid=="S9957-D012"] <- TRUE

# Check
letters  %>% 
filter(docauthorname=="Hutchings, James Mason, 1820-1902") %>%
select(docid, occupation, publicLetter)  %>% 
unique()

docid      occupation                            publicLetter
1 S9957-D012 Tradesman; Miner; Writer; Businessman TRUE        
2 S9957-D013 Tradesman; Miner; Writer; Businessman   NA        
3 S9957-D014 Tradesman; Miner; Writer; Businessman   NA        
4 S9957-D015 Tradesman; Miner; Writer; Businessman   NA

In [393]:
# Line 255
letters  %>% 
filter(docauthorname=="Downe, John, fl. 1830") %>%
select(docid, nationalOrigin, religionNew, relMin, authorLocation, native_occupation, north_american_occupation, publicLetter)  %>% 
unique()

docid      nationalOrigin religionNew relMin authorLocation native_occupation
1 S9974-D010 English        NA          NA     USA            Tradesman        
  north_american_occupation publicLetter
1 Manufacturer              NA

"Obviously this weaver, who seems to have had no difficulty in becoming the manager of a Hudson Valley factory shortly after his arrival"

Because he seems to have been elevated to a management position by the time of writing, I will not recode occupation to include the native occupation.

In [394]:
# Now compare vairables
letters %>% #in the dataframe
filter(row_number() %in% rows) %>% # in the rows that meet those conditions
select(docauthorname, native_occupation, north_american_occupation, occupation) %>% # view values for these variables
unique() # only once where they occur together

docauthorname                      native_occupation            
1   Harris, Critchlow, 1813-1899       Businessman; Farmer; Merchant
2   Harris, Sarah Stretch, 1818-1897   Businessman; Educator        
41  Harris, Tom, 1847-                 Businessman; Farmer; Merchant
51  Harris, Robert, 1849-1919          Businessman; Farmer; Merchant
81  Harris, Margaret Ellin, 1851-1944  Businessman; Farmer; Merchant
176 Harris, William, 1854-             Businessman; Farmer; Merchant
228 Carrothers, Nathaniel, ?-1881      Farmer                       
235 Robb, Alexander, 1839-             Farmer                       
247 Bishop, Robert Hamilton, 1777-1855 Clergy                       
248 McKenzie, D., fl. 1834             Clergy                       
249 Aylmer, Matthew, Lord, 1775-1850   Military personnel           
250 Singer, William, fl. 1831          Tradesman                    
251 Hutchings, James Mason, 1820-1902  Tradesman                    
255 Downe, John, fl. 1830              Tradesman                    
    north_american_occupation                                         
1   Architect; Artist; Businessman; Clergy; Educator                  
2   Architect; Artist; Businessman; Clergy                            
41  Architect; Artist; Businessman; Clergy; Educator; Farmer          
51  Architect; Artist; Businessman; Clergy; Educator; Farmer          
81  Architect; Artist; Businessman; Educator; Farmer                  
176 Architect; Artist; Businessman; Clergy; Educator; Farmer; Merchant
228 Farmer; Tradesman                                                 
235 Miner; Rancher; Laborer                                           
247 Clergy; Educator; Writer                                          
248 Clergy; Missionary                                                
249 Royal governor                                                    
250 Laborer; Tradesman                                                
251 Miner                                                             
255 Manufacturer                                                      
    occupation                                  
1   Farmer; Merchant; Laborer                   
2   Farmer; Merchant; Laborer                   
41  Businessman                                 
51  Artist; Educator                            
81  Writer; Printer                             
176 Architect                                   
228 Farmer; Tradesman                           
235 Miner; Laborer; Rancher; Government employee
247 Clergy; Educator; Writer                    
248 Clergy; Missionary                          
249 Military personnel; Royal governor          
250 Laborer; Tradesman                          
251 Tradesman; Miner; Writer; Businessman       
255 Manufacturer

## Irish / Irish; Scottish and religion

In [395]:
letters %>% 
filter(nationalOrigin=="Irish" | 
       nationalOrigin=="Irish; Scottish") %>% 
select(nationalOrigin, docauthorname, religionNew, relMin, publicLetter) %>% 
unique()

nationalOrigin  docauthorname                           religionNew relMin
1  Irish           Humphrey, James, fl. 1824               NA             NA 
2  Irish           Mee, Patrick, fl. 1844                  Catholic     TRUE 
3  Irish; Scottish Carrothers, Nathaniel, ?-1881           Anglican    FALSE 
10 Irish; Scottish Carrothers, Joseph, 1793(?)-            Methodist   FALSE 
21 Irish           White, Jane, 1831(?)-1867               NA             NA 
27 Irish           Robb, Alexander, 1839-                  NA             NA 
39 Irish; Scottish Buchanan, J. C., fl. 1833               NA             NA 
40 Irish; Scottish Buchanan, Alexander Carlisle, 1786-1840 NA             NA 
41 Irish           Graham, Thomas, fl. 1827                NA             NA 
43 Irish           Sampson, William, 1764-1836             NA             NA 
53 Irish           Mooney, Thomas, fl. 1850                NA             NA 
   publicLetter
1  NA          
2  NA          
3  NA          
10 NA          
21 NA          
27 NA          
39 NA          
40 NA          
41 NA          
43 NA          
53 NA

In [396]:
letters  %>% 
filter(docauthorname=="Humphrey, James, fl. 1824") %>%
select(sourcetitle, nationalOrigin, religionNew, relMin, authorLocation)  %>% 
unique()

sourcetitle                                                           
1 Irish Emigration and Canadian Settlement: Patterns, Links, and Letters
  nationalOrigin religionNew relMin authorLocation
1 Irish          NA          NA     Canada

Impossible to say what his relgious affiliation is. He appears to be from Derry in the North of Ireland. His family name seems to be more associated with English / Scottish (i.e., Protestant) lineages than Irish Catholic ones. In the original text (S9635-D006), he makes references to a large social network that includes people with family names of Gaelic and non-Gaelic origins. Given this high degree of uncertainty, I will leave the religion variables alone.
https://en.wikipedia.org/wiki/Derry
https://coadb.com/surnames/humphrey-arms.html

In [397]:
letters  %>% 
filter(docauthorname=="Mee, Patrick, fl. 1844") %>%
select(docid, nationalOrigin, religionNew, relMin, authorLocation)  %>% 
unique()

docid      nationalOrigin religionNew relMin authorLocation
1 S9635-D010 Irish          Catholic    TRUE   Canada

In [398]:
letters  %>% 
filter(docauthorname=="White, Jane, 1831(?)-1867") %>%
select(sourcetitle, nationalOrigin, religionNew, relMin, authorLocation, occupation)  %>% 
unique()

sourcetitle                                                           
1 Irish Emigration and Canadian Settlement: Patterns, Links, and Letters
  nationalOrigin religionNew relMin authorLocation occupation
1 Irish          NA          NA     Canada         NA

According to the original source, he was from a "bourgeois" family in County Down in Northern Ireland. Her father's main source of income was from rents on property he owned. He is described as a minor real estate speculator who had a modest income. The editor of the volume in which Jane's letters reprinted speculates that the family came to Canada with considerable funds, given the father's ability to buy land, their class during the voyage, and the fact that they brought Jane's paino (recall Getz saying this was the ultimate sign of middle class status). The quality of Jane's writing indicates to the editor that her father had been a merchant or small manufacturer. She isn described as "Protestant Irish".

In [399]:
letters$relMin[letters$docauthorname=="White, Jane, 1831(?)-1867"]  <- FALSE
letters$occupation[letters$docauthorname=="White, Jane, 1831(?)-1867"]  <- "Businessman"

In [400]:
letters  %>% 
filter(docauthorname=="Robb, Alexander, 1839-") %>%
select(sourcetitle, nationalOrigin, religionNew, relMin, authorLocation, occupation)  %>% 
unique()

sourcetitle                                                           
1 Irish Emigration and Canadian Settlement: Patterns, Links, and Letters
  nationalOrigin religionNew relMin authorLocation
1 Irish          NA          NA     Canada        
  occupation                                  
1 Miner; Laborer; Rancher; Government employee

In [401]:
letters$relMin[letters$docauthorname=="Robb, Alexander, 1839-"] <- FALSE

In [402]:
# Buchanan, J. C., fl. 1833
letters  %>% 
filter(docauthorname=="Buchanan, J. C., fl. 1833") %>%
select(docid, nationalOrigin, religionNew, relMin, authorLocation, occupation)  %>% 
unique()

docid      nationalOrigin  religionNew relMin authorLocation
1 S9865-D009 Irish; Scottish NA          NA     USA           
  occupation         
1 Government employee

This is James Clarke Buchanan (1801 - 1849), son of James Buchanan (1772 - 1851).
<ul>
    <li>http://www.ccchs.ca/Newsletters/CCCHS%20Newsletter--Winter-Spring%202019%20.pdf)</li>
    <li>https://www.findagrave.com/memorial/90711480/james-clarke-buchanan</li>
<li>https://archive.org/details/galtscanadianody0001timo/page/100/mode/2up?q=%22James+Clarke+Buchanan%22</li>
    </ul>
His father was baptiste, as described in the text block below, which would make the son not from a religious minority, although his current religious affiliation is unknown.

In [403]:
letters$relMin[letters$docauthorname=="Buchanan, J. C., fl. 1833"]  <- FALSE

In [404]:
# Buchanan, Alexander Carlisle, 1786-1840
letters  %>% 
filter(docauthorname=="Buchanan, Alexander Carlisle, 1786-1840") %>%
select(docid, nationalOrigin, religion, religionNew, relMin, authorLocation, docyear, occupation)  %>% 
unique()

docid      nationalOrigin  religion religionNew relMin authorLocation docyear
1 S9865-D010 Irish; Scottish NA       NA          NA     Canada         1834   
  occupation          
1 Government appointee

Alexander Carlisle Buchanan is the brother of James Buchanan, who was brought up Presbyterian but converted to Baptist. This would make him part of the religious majority, although it cannot be said what his denomination was. 
http://brixton38.biographi.ca/en/bio/buchanan_alexander_carlisle_1786_1840_7E.html
http://www.ccchs.ca/Newsletters/CCCHS%20Newsletter--Winter-Spring%202019%20.pdf

In [405]:
letters$relMin[letters$docauthorname=="Buchanan, Alexander Carlisle, 1786-1840"]  <- FALSE

In [406]:
# Graham, Thomas, fl. 1827
letters  %>% 
filter(docauthorname=="Graham, Thomas, fl. 1827") %>%
select(sourcetitle, nationalOrigin, religion, religionNew, relMin, authorLocation, occupation)  %>% 
unique()

sourcetitle                                                                                                           
1 Hints on Emigration to Upper Canada; Especially Addressed to the Middle and Lower Classes in Great Britain and Ireland
  nationalOrigin religion religionNew relMin authorLocation occupation
1 Irish          NA       NA          NA     Canada         NA

The texts give no indication of the author's religion; however, his surname suggests greater affiliation with Irish / Scottish / Protestant traditions than Irish-Catholic ones. His occupation is given as a joiner, and he refers to an apprenticeship, meaning that this is a skilled trade. As such, he will be coded as a tradesman.
https://en.wikipedia.org/wiki/County_Monaghan

In [407]:
# Note this in case it makes sense to omit public letters at a later stage.
letters$occupation[letters$docauthorname=="Graham, Thomas, fl. 1827"] <- "Tradesman"

One letter (S9873-D014) is addressed to his mother Elizabeth in Enniscorthy Ireland, which is in the north. Given that the family name Graham has Scottish and English origins, it is likely that this writer's religious affiliation is Protestant, and hence not a religious minority. However, this town does have Catholic churches and of course conversion even across the Protestant - Catholic divide was possible. I will therefore not recode any of the relgion vairables. https://en.wikipedia.org/wiki/Enniscorthy

William Sampson is an interesting case. He was an Irish lawyer who was exiled because of his political (anti Union) activism. The letters are lengthy, highly political and published as part of his memoir.

In [408]:
# Note this in case it makes sense to omit public letters at a later stage.
letters$publicLetter[letters$docauthorname=="Sampson, William, 1764-1836"] <- TRUE

He was of a Protestant background although it is unclear which denomination. Accounts show his father as an Anglican clergyman, so it is safe to say his cultural background is not that of a religion minority. Again, it is possible that he converted during his life. With this in mind, I will recode him as FALSE for relMin but leave the religionNew value as NA because his denomination is uncertain. 
https://www.dib.ie/biography/sampson-william-a7907
https://history.nycourts.gov/figure/william-sampson/

In [409]:
letters$relMin[letters$docauthorname=="Sampson, William, 1764-1836"] <- FALSE

In [410]:
# Mooney, Thomas, fl. 1850
letters  %>% 
filter(docauthorname=="Mooney, Thomas, fl. 1850") %>%
select(sourcetitle, nationalOrigin, religion, religionNew, relMin, occupation)  %>% 
unique()

sourcetitle           nationalOrigin religion religionNew relMin occupation
1 Nine Years In America Irish          NA       NA          NA     Writer

The Irish family name Mooney is derived from Gaelic. This plus the writer's anti-British, pro-Irish attitudes seem to indicate that his cultural background is more likely Irish-Catholic than Irish-Protestant. The description of him as an "eccentric spiritualist" indicates that he might not have been a traditional Catholic, but then he probably was also never part of the Protestant majority.
https://en.wikipedia.org/wiki/Mooney
https://www.dib.ie/biography/mooney-thomas-a5917
https://archive.org/details/historyofireland01moon/page/n24/mode/1up?q=Catholic

In [411]:
# Note this in case it makes sense to omit public letters at a later stage.
letters$publicLetter[letters$docauthorname=="Mooney, Thomas, fl. 1850"] <- TRUE
letters$relMin[letters$docauthorname=="Mooney, Thomas, fl. 1850"] <- TRUE

All of the people identified as "Irish; Scottish" originated in Ireland, even if their ancestors and identities were Scottish. For the purpose of the modeling I intend to do, their actual national origin plus their status as a religious minority within the context of the British empire (i.e., Protestant or not) is what matters. With this in mind, I am recodings nationalOrigin as Irish for these individuals. 

In [412]:
letters %>% 
filter(nationalOrigin=="Irish; Scottish") %>% 
select(cultural_heritage, nationalOrigin, docauthorname, religionNew, relMin) %>% 
unique()

cultural_heritage         nationalOrigin 
1  Irish; European; Scottish Irish; Scottish
8  Irish; European; Scottish Irish; Scottish
19 NA                        Irish; Scottish
20 NA                        Irish; Scottish
   docauthorname                           religionNew relMin
1  Carrothers, Nathaniel, ?-1881           Anglican    FALSE 
8  Carrothers, Joseph, 1793(?)-            Methodist   FALSE 
19 Buchanan, J. C., fl. 1833               NA          FALSE 
20 Buchanan, Alexander Carlisle, 1786-1840 NA          FALSE

Note that the cultural_heritage value for J.C. and A.C. Buchanan was NA. The Irish Scottish designation for came from research for done to recode the nationalOrigin variable (see Nationality <a href="https://nbviewer.org/github/menyalas/dissertation/blob/main/20230217_AM_PhD-NaildohNationality.ipynb">notebook</a> and <a href="https://wellbeinginlifewriting.wordpress.com/2023/02/17/naldoh-nationality/">blog post</a>).

In [413]:
# Now recode nationalOrigin to Irish 
letters$nationalOrigin[letters$nationalOrigin=="Irish; Scottish"] <- "Irish"

In [414]:
letters %>% 
filter(nationalOrigin=="Irish") %>% 
select(docauthorname, religionNew, relMin, publicLetter) %>% 
unique()

docauthorname                           religionNew relMin publicLetter
1  Humphrey, James, fl. 1824               NA             NA    NA        
2  Mee, Patrick, fl. 1844                  Catholic     TRUE    NA        
3  Carrothers, Nathaniel, ?-1881           Anglican    FALSE    NA        
10 Carrothers, Joseph, 1793(?)-            Methodist   FALSE    NA        
21 White, Jane, 1831(?)-1867               NA          FALSE    NA        
27 Robb, Alexander, 1839-                  NA          FALSE    NA        
39 Buchanan, J. C., fl. 1833               NA          FALSE    NA        
40 Buchanan, Alexander Carlisle, 1786-1840 NA          FALSE    NA        
41 Graham, Thomas, fl. 1827                NA             NA    NA        
43 Sampson, William, 1764-1836             NA          FALSE  TRUE        
53 Mooney, Thomas, fl. 1850                NA           TRUE  TRUE

In [415]:
write.csv(letters, 
          "20230423_AM_PhD-NaildohSubset.csv", 
          row.names=FALSE)